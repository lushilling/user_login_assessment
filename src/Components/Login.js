import React from 'react';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import axios from 'axios';

export default class Login extends React.Component {

    constructor() {
        super();
        this.state = {
            username: "",
            password: ""
        }
    }
    getRequest = (e) => {
        e.preventDefault();

        let loginUser = {
            username: e.target[0].value,
            password: e.target[1].value
        }

        if (loginUser.username === "" || loginUser.password === "") {
            this.setState({ error: "Please ensure you have filled out all the fields" })
        } else {
            this.setState({ error: "" })
        }

        axios.get("http://localhost:5000/user/name/" + loginUser.username + "/" + loginUser.password)
            .then(response => {
                this.setState({
                    data: response.data
                });
            });
    }
    render() {
        return (
            <div>
                <Form inline onSubmit={this.getRequest}>
                    <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
                        <Label for="username" className="mr-sm-2">Username: </Label>
                        <Input type="text" name="username" id="username" placeholder="Enter Username" />
                    </FormGroup>
                    <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
                        <Label for="password" className="mr-sm-2">Password: </Label>
                        <Input type="password" name="password" id="password" placeholder="Enter Password" />
                    </FormGroup>
                    <Button>Login</Button>
                </Form>
                <p style={{ color: 'red' }}>{this.state.error}</p>
            </div>
        );
    }
}