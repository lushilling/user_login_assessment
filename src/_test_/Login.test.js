import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from "react-router-dom";

import Login from '../Components/Login';


it('renders App', () => {
    const tree = renderer.create(<Login/>).toJSON()
    expect(tree).toMatchSnapshot();
  })


// it('renders router without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<Router />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

// it('renders navigation bar without crashing', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(<NavigationBar />, div);
//     ReactDOM.unmountComponentAtNode(div);
//   });

it('renders app without crashing', () => {
  shallow(<Login />);
});

//Testing rendering a Prop
it("render a prop", () => {
  const wrapper = shallow(<Login username="A Name" />);
  expect(wrapper.instance().props.username).toEqual("A Name");
  expect(wrapper).toMatchSnapshot();
});