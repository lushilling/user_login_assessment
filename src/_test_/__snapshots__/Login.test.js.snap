// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a prop 1`] = `ShallowWrapper {}`;

exports[`renders App 1`] = `
<div>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="username"
      >
        Username: 
      </label>
      <input
        className="form-control"
        id="username"
        name="username"
        placeholder="Enter Username"
        type="text"
      />
    </div>
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Enter Password"
        type="password"
      />
    </div>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
